pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -331,6 +339,64 @@ func (c *EtcdCertSignerController) createSecretForNode(node *corev1.Node) error
 	return nil
 }
 
+func (c *EtcdCertSignerController) Get43CompatibilityNames(node *corev1.Node) []string {
+	var errs []string
+	etcdDiscoveryDomain, err := c.getEtcdDiscoveryDomain()
+	if err != nil {
+		errs = append(errs, err.Error())
+	}
+	nodeInternalIPs, err := dnshelpers.GetInternalIPAddressesForNodeName(node)
+	if err != nil {
+		errs = append(errs, err.Error())
+	}
+	etcdMembers, err := c.etcdClient.MemberList()
+	if err != nil {
+		errs = append(errs, err.Error())
+	}
+	peerHostNames, err := getPeerDNSSubjectAlternativeNamesFromMemberList(etcdDiscoveryDomain, etcdMembers, nodeInternalIPs)
+	if err != nil {
+		errs = append(errs, err.Error())
+	}
+
+	// fallback to wildcard
+	if len(errs) > 0 {
+		return []string{"*." + etcdDiscoveryDomain}
+	}
+	return peerHostNames
```

Comment:
is this for 4.3 or 4.4?

Reply:
if we observe no 4.3 naming we return nil otherwise we return explicit DNS records for 4.3 SAN